### PR TITLE
refactor(mcp-server): unify daemon.json validation on a single Zod schema

### PR DESCRIPTION
**Pull request by kamiazya**
## Summary

`daemon-registry.ts` and `daemon-record.ts` each carried a hand-written validator for `daemon.json` — the security-sensitive record holding the Bearer token for all HTTP/WS auth — and the two already disagreed on missing-token handling (`loadDaemonRecord` accepted `token: ''` while `parseDaemonRecord` classified it token-missing). This is a direct violation of the repo's Zod-single-source rule (every other persisted JSON goes through `safeParse`).

- New `daemon-record-schema.ts` defines `daemonRecordSchema` once; `DaemonRecord` is derived via `z.infer` (no parallel hand-written interface). Placed in its own module to avoid a `daemon-record.ts → daemon-registry.ts` import cycle.
- Both `loadDaemonRecord` and `parseDaemonRecord` validate via `safeParse`; the two typeof-chains are deleted.
- **Missing/empty token unified fail-closed**: `token: ''` or absent is now rejected by both paths.
- `pid`/`port` tightened to positive integers, `port` capped at 65535 (what the daemon actually writes). Unknown keys still stripped for forward compatibility.

## Test plan

- [x] Red-first: well-formed round-trip parse; token-missing/empty rejected; wrong-typed fields (`pid`/`port` as string, non-object, invalid JSON) rejected
- [x] Mutation check: loosening `token` to `z.string()` (no `.min(1)`) turns the empty-token test red
- [x] `pnpm test --project mcp-node` daemon suites green (69 tests); `pnpm --filter @kamiazya/whiteboard-mcp typecheck` clean
- [x] No plaintext secrets in fixtures/logs (token stays a fake placeholder; static parse messages, no value interpolation)

🤖 Generated with [Claude Code](https://claude.com/claude-code)

## Summary by CodeRabbit

- **New Features**
  - Added schema-based validation for stored daemon records, including required fields, authentication tokens, positive PID, and TCP port bounds (1–65535).
  - Extra unknown fields are ignored while preserving known valid values.

- **Bug Fixes**
  - Empty or missing authentication tokens now fail closed (treated as invalid) rather than being accepted.
  - Non-object or malformed records are now rejected safely with consistent behavior.

- **Tests**
  - Expanded coverage for valid, malformed, incomplete, and invalid daemon record inputs and edge cases.

### CODE REVIEW

File: packages/mcp-server/src/daemon/daemon-record.ts [hunk 1]
```diff
@@ -28,30 +34,22 @@ export async function parseDaemonRecord(dataDir: string): Promise<DaemonRecordPa
...
   }
 
-  if (typeof obj.token !== 'string' || obj.token === '') {
-    return { kind: 'token-missing', record: base }
+  if (typeof parsed !== 'object' || parsed === null) {
+    return { kind: 'malformed', message: 'Daemon record is not an object.' }
   }
 
-  return { kind: 'valid', record: { ...base, token: obj.token } }
+  return { kind: 'malformed', message: 'Daemon record is missing required fields.' }
```
[the remark this answers is not in the document]
Comment (kamiazya) [reply packages/mcp-server/src/daemon/daemon-record.ts:59]: Fixed — arrays are now rejected up front with a clear 'not an object' message, and a present-but-wrong-typed token (e.g. token:123) is classified malformed instead of slipping through the token-stripping base schema as token-missing. Added both cases to daemon-record.test.ts.

File: packages/mcp-server/src/daemon/daemon-record.ts [hunk 2]
```diff
@@ -28,30 +34,22 @@ export async function parseDaemonRecord(dataDir: string): Promise<DaemonRecordPa
...
-    port: obj.port,
-    version: obj.version,
-    startedAt: obj.startedAt,
+  // The full schema also fails for missing/empty token, so re-check against
+  // the token-less base schema to distinguish "token missing" (a record we
+  // can still report on) from genuinely malformed data.
+  const baseResult = daemonRecordBaseSchema.safeParse(parsed)
+  if (baseResult.success) {
+    return { kind: 'token-missing', record: baseResult.data }
   }
```
[the remark this answers is not in the document]
Comment (kamiazya) [reply packages/mcp-server/src/daemon/daemon-record.ts:57]: Fixed — arrays are now rejected up front with a clear 'not an object' message, and a present-but-wrong-typed token (e.g. token:123) is classified malformed instead of slipping through the token-stripping base schema as token-missing. Added both cases to daemon-record.test.ts.
